pandas/io/pickle.py

```diff
@@ -32,12 +44,21 @@ def read_pickle(path):
     ----------
     path : string
```

Comment:
It shouldn't be too hard to get `read_pickle` to also support reading from URLs. @goldenbull -- not sure if you are also interested in adding this feature. Ultimately all read methods should support compression and URL reading.